Automattic/wp-calypso
# Update/move signup stepheader to blocks

StepHeader will be required by two modules, jetpack-connect and signup (#14321). This PR moves it to `blocks`.

To test:
* Use this branch
* Follow flows using `StepHeader`
  * http://calypso.localhost:3000/jetpack/connect/store
  * http://calypso.localhost:3000/start
* Ensure code compiles correctly, tests pass, and there are no regressions or console errors/warnings

Supersedes #14341 which had to be reverted due to e2e test conflicts.
Blocked by https://github.com/Automattic/wp-e2e-tests/issues/553
